packages/core/src/connectors/facebook/index.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * Reference: Manually Build a Login Flow
+ * https://developers.facebook.com/docs/facebook-login/manually-build-a-login-flow
+ */
+import got, { RequestError as GotRequestError } from 'got';
+import { stringify } from 'query-string';
+import { z } from 'zod';
+
+import {
+  accessTokenEndpoint,
+  authorizationEndpoint,
+  scope,
+  userInfoEndpoint,
+} from '@/connectors/facebook/constant';
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  ConnectorMetadata,
+  ConnectorType,
+  GetAccessToken,
+  GetAuthorizationUri,
+  GetUserInfo,
+  ValidateConfig,
+} from '@/connectors/types';
+import { getConnectorConfig, getConnectorRequestTimeout } from '@/connectors/utilities';
+
+export const metadata: ConnectorMetadata = {
+  id: 'facebook',
+  type: ConnectorType.Social,
+  name: {
+    en: 'Sign In with Facebook',
+    zh_CN: 'Facebook 登录',
+  },
+  logo: './logo.png',
+  description: {
+    en: 'Sign In with Facebook',
+    zh_CN: 'Facebook 登录',
+  },
+};
+
+const facebookConfigGuard = z.object({
+  clientId: z.string(),
+  clientSecret: z.string(),
+});
+
+type FacebookConfig = z.infer<typeof facebookConfigGuard>;
+
+export const validateConfig: ValidateConfig = async (config: unknown) => {
+  if (!config) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, 'Missing config');
+  }
+
+  const result = facebookConfigGuard.safeParse(config);
+
+  if (!result.success) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, result.error.message);
+  }
+};
+
+export const getAuthorizationUri: GetAuthorizationUri = async (redirectUri, state) => {
+  const config = await getConnectorConfig<FacebookConfig>(metadata.id);
+
+  return `${authorizationEndpoint}?${stringify({
+    client_id: config.clientId,
+    redirect_uri: redirectUri,
+    response_type: 'code',
```

Comment:
as `response_type` is fixed to be `code`, should we define `response_type` as a constant like `scope`?

Reply:
Since we just only need `code` response type for now, better keep it and add it when needed?

Any idea? @logto-io/eng 

---
(If needed, plan to extract it as a constant like this in another PR.)
<img width="1464" alt="image" src="https://user-images.githubusercontent.com/10594507/156979095-3991ffbb-2640-4b84-8cb1-9f55b32a6b44.png">

Reply:
it should be an enum at least for FB connector? not sure the value of `response_type` for other potential social connectors.
<img width="910" alt="image" src="https://user-images.githubusercontent.com/15182327/157007648-02885079-7aea-46a4-8b81-b1169eee3725.png">
attached is the link FYI: https://developers.facebook.com/docs/facebook-login/manually-build-a-login-flow

Reply:
maybe add an issue to do a little research on this.

Reply:
Is OK to keep it.

Reply:
Conclusion: no need.
See [discussion](https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1646708368080299) on Slack.